postgres/check.py

```diff
@@ -606,7 +606,7 @@ def _collect_stats(self, key, db, instance_tags, relations, custom_metrics, func
                             config_schemas = relations_config[relname]['schemas']
                             if config_schemas and desc_map['schema'] not in config_schemas:
                                 continue
-                        except KeyError:
+                        except (KeyError,UnboundLocalError):
```

Comment:
This is tricky because makes the code harder to read (it's not obvious that `relation_config` doesn't exist, it might be any other variable within the `try`).
What about changing line 603 to `if 'schema' in desc_map and relations:` ?